30411/Alungei C. Paula Carla/lab1.txt

```diff
@@ -0,0 +1,229 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct node
+{
+    int code;
+    struct node* next;
+}NodeT;
+
+NodeT *head;
+NodeT *tail, *now, *next;
+
+void addFirst(int code)
+{
+    if (head==NULL)
+    {
+        head=(NodeT*)malloc (sizeof (NodeT));
+        head ->code=code;
+        head->next=NULL;
+    }
+    else if (tail==NULL)
```

Comment:
we can somehow eliminate this second if branch.
try putting ```tail=head``` in the first if branch as a last statement and remove this whole branch entirely. Does it work? why?